tests/draft2020-12/unevaluatedProperties.json

```diff
@@ -715,6 +715,60 @@
             }
         ]
     },
+    {
+        "description": "unevaluatedProperties with $dynamicRef",
+        "schema": {
+            "$schema": "https://json-schema.org/draft/2020-12/schema",
+            "$id": "https://example.com/derived",
+
+            "$ref": "/baseSchema",
+
+            "$defs": {
+                "derived": {
+                    "$dynamicAnchor": "addons",
+                    "properties": {
+                        "bar": { "type": "string" }
+                    }
+                },
+                "baseSchema": {
+                    "$id": "/baseSchema",
+
+                    "type": "object",
+                    "properties": {
+                        "foo": { "type": "string" }
+                    },
+                    "$dynamicRef": "#addons",
+                    "unevaluatedProperties": false,
```

Comment:
The order of keywords might affect order of execution in implementations, so should we put `unevaluatedProperties` first, to test that as well?

Reply:
We have other tests that ensure key order is not a factor in evaluation.

Reply:
Not for `$dynamicRef` and `$ref` though. My implementation passed this test case but just by luck 😅 

Moving `"$ref": "#/$defs/bar"` down in `"unevaluatedItems with $ref"`, for example, fails.

It's also really important to test references bc they are not explicitly listed in the spec section: https://json-schema.org/draft/2020-12/json-schema-core#name-unevaluatedproperties

Reply:
> Not for $dynamicRef and $ref though.

I don't think it's practical to check order of keywords for every combination of keywords.  We have tests that ensure order of keywords is irrelevant. It's expected that this is applied across an implementation, not just for those tests.

Reply:
> It's also really important to test references bc they are not explicitly listed in the spec section: https://json-schema.org/draft/2020-12/json-schema-core#name-unevaluatedproperties

The spec does address references:

"This includes but is not limited to the in-place applicators defined in this document."

References are in-place applicators because they evaluate the instance against the schema they're referencing.

Reply:
While I agree that it's impractical to test ordering for every combination of keywords, I'm of the opinion that anything that exposed a real bug in an implementation belongs in the test suite. I'll move `unevaluated*` to be the first keyword and include a note explaining why.